cdap-app-templates/cdap-etl/cdap-etl-realtime/src/main/java/co/cask/cdap/templates/etl/realtime/ETLRealtimeTemplate.java

```diff
@@ -50,6 +51,7 @@ public ETLRealtimeTemplate() throws Exception {
   public void configureAdapter(String adapterName, ETLRealtimeConfig etlConfig, AdapterConfigurer configurer)
     throws Exception {
     super.configureAdapter(adapterName, etlConfig, configurer);
+    Preconditions.checkArgument(etlConfig.getInstances() != 0, "Instances cannot be 0");
```

Comment:
By this time, has default 0 has been applied right (scenario - its unspecified in configuration). Otherwise you may have to check for null?